src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -253,58 +274,109 @@ pub struct RSIndexResult {
+            result_type: RSResultType::Virtual,
+            is_copy: false,
+            metrics: ptr::null_mut(),
+            weight,
+        }
+    }
+
+    /// Create a new numeric index result with the given numeric value
+    pub fn numeric(doc_id: t_docId, num: f64) -> Self {
+        Self {
+            doc_id,
+            field_mask: RS_FIELDMASK_ALL,
+            freq: 1,
             data: RSIndexResultData {
                 num: ManuallyDrop::new(RSNumericRecord(num)),
             },
             result_type: RSResultType::Numeric,
-            is_copy: false,
-            metrics: std::ptr::null_mut(),
-            weight: 0.0,
+            weight: 1.0,
+            ..Default::default()
         }
     }
 
-    /// Create a new virtual index result
-    pub fn virt(doc_id: t_docId, field_mask: FieldMask, weight: f64) -> Self {
+    /// Create a new metric index result with the given document ID
+    pub fn metric(doc_id: t_docId) -> Self {
         Self {
             doc_id,
-            dmd: std::ptr::null(),
-            field_mask,
-            freq: 0,
-            offsets_sz: 0,
+            field_mask: RS_FIELDMASK_ALL,
             data: RSIndexResultData {
-                virt: ManuallyDrop::new(RSVirtualResult),
+                num: ManuallyDrop::new(RSNumericRecord(0.0)),
             },
-            result_type: RSResultType::Virtual,
-            is_copy: false,
-            metrics: std::ptr::null_mut(),
+            result_type: RSResultType::Metric,
+            weight: 1.0,
+            ..Default::default()
+        }
+    }
+
+    /// Create a new intersection index result with the given document ID, capacity, and weight
+    pub fn intersect(doc_id: t_docId, cap: usize, weight: f64) -> Self {
+        Self {
+            doc_id,
+            data: RSIndexResultData {
+                agg: ManuallyDrop::new(RSAggregateResult::with_capacity(cap)),
+            },
+            result_type: RSResultType::Intersection,
             weight,
+            ..Default::default()
         }
     }
 
     /// Create a new union index result with the given document ID, capacity, and weight
     pub fn union(doc_id: t_docId, cap: usize, weight: f64) -> Self {
         Self {
             doc_id,
-            dmd: std::ptr::null(),
-            field_mask: 0,
-            freq: 0,
-            offsets_sz: 0,
             data: RSIndexResultData {
                 agg: ManuallyDrop::new(RSAggregateResult::with_capacity(cap)),
             },
             result_type: RSResultType::Union,
-            is_copy: false,
-            metrics: std::ptr::null_mut(),
             weight,
+            ..Default::default()
+        }
+    }
+
+    /// Create a new hybrid metric index result with the given document ID
+    pub fn hybrid_metric(doc_id: t_docId) -> Self {
+        Self {
+            doc_id,
+            data: RSIndexResultData {
+                agg: ManuallyDrop::new(RSAggregateResult::with_capacity(2)),
```

Comment:
why 2?

Reply:
It's just based on the C version, but I'm not sure why it is 2 :sweat_smile: This reminds me of something I saw earlier in the week which seemed to indicate that a hybrid metric only ever has 2 entries. I'll reach out to Guy to confirm